src/server/game/Spells/SpellInfo.cpp

```diff
@@ -1549,7 +1549,24 @@ bool SpellInfo::IsRequiringDeadTarget() const
 
 bool SpellInfo::IsAllowingDeadTarget() const
 {
-    return HasAttribute(SPELL_ATTR2_CAN_TARGET_DEAD) || Targets & (TARGET_FLAG_CORPSE_ALLY | TARGET_FLAG_CORPSE_ENEMY | TARGET_FLAG_UNIT_DEAD);
+    if (HasAttribute(SPELL_ATTR2_CAN_TARGET_DEAD) || Targets & (TARGET_FLAG_CORPSE_ALLY | TARGET_FLAG_CORPSE_ENEMY | TARGET_FLAG_UNIT_DEAD))
+    {
+        return true;
+    }
+
+    for (SpellEffectInfo const* effect : _effects)
+    {
+        if (!effect)
+            continue;
+
+        switch (effect->TargetA.GetTarget())
```

Comment:
Replace the switch with `if (effect->TargetA.GetObjectType() == TARGET_OBJECT_TYPE_CORPSE)`
And add TargetB

Reply:
The reason I used a switch is because I saw few more enum values with `CORPSE` in them, but wasn't sure which I should include

Are you saying I shouldn't include anything else?

Reply:
The reason I used a switch is because I saw few more enum values with CORPSE in them, but wasn't sure which I should include

Are you saying I shouldn't include anything else?

Reply:
Look closer, I'm suggesting replacing `GetTarget()` with `GetObjectType()`

Reply:
@Shauren my bad.

Added TargetB and also using `GetObjectType()` now like you suggested.